tests/common/quantization/data_generators.py

```diff
@@ -0,0 +1,372 @@
+
+    result = np.sort(np.concatenate([result, interval_points]))
+    is_near_mid_point = get_points_near_of_mid_points(result, quant_mid_points, quant_len * rtol_for_mid_point)
+    quant_lens = np.repeat(quant_len, num_elements)
+
+    input_data = result.reshape(input_size)
+    is_near_mid_point = is_near_mid_point.reshape(input_size)
+    quant_lens = quant_lens.reshape(input_size)
+
+    return input_data, is_near_mid_point, quant_lens
+
+
+def generate_test_input(
+    input_size: Tuple,
+    input_low: np.array,
+    input_range: np.array,
+    bits: int,
+    is_per_channel: bool,
+    is_weights: bool,
+    rtol_for_mid_point: float = 0.00005,
+) -> np.array:
+    """
+    Generate test input data.
+
+    :param input_size: Size of input tensor.
+    :param input_low: Array with low values.
+    :param input_range: Array with range values.
+    :param bits: Number of bits of quantization.
+    :param is_per_channel: `True` for per-channel quantization, `False` for per-tensor.
+    :param is_weights: Boolean that defines tensor type. True for Weights, False for Activations.
+    :param rtol_for_mid_point: Relative tolerant value for points is in the middle between quants, defaults to 0.00005.
+
+    :return np.array: inputs, is_near_mid_point, quant_lens
+    """
+    if not is_per_channel:
+        return generate_one_channel_input(input_low, input_range, input_size, bits, rtol_for_mid_point)
+
+    inputs = None
+    is_near_mid_point = None
+    quant_lens = None
+    if is_per_channel:
+        if is_weights:
+            channel_count = input_size[0]
+            inputs = np.empty(input_size)
+            is_near_mid_point = np.zeros(input_size).astype(np.bool)
+            quant_lens = np.empty(input_size)
+            for idx in range(0, channel_count):
+                ch_input, ch_is_near_mid_point, ch_quant_lens = generate_one_channel_input(
+                    input_low[idx], input_range[idx], input_size[1:], bits, rtol_for_mid_point
+                )
+                inputs[idx] = ch_input
+                is_near_mid_point[idx] = ch_is_near_mid_point
+                quant_lens[idx] = ch_quant_lens
+        else:
+            channel_count = input_size[1]
+            inputs = np.empty(input_size)
+            is_near_mid_point = np.zeros(input_size).astype(np.bool)
+            quant_lens = np.empty(input_size)
+            for idx in range(0, channel_count):
+                ch_input, ch_is_near_mid_point, ch_quant_lens = generate_one_channel_input(
+                    input_low[idx], input_range[idx], input_size[0:1] + input_size[2:], bits, rtol_for_mid_point
+                )
+                inputs[:, idx] = ch_input
+                is_near_mid_point[:, idx] = ch_is_near_mid_point
+                quant_lens[:, idx] = ch_quant_lens
+
+    return inputs, is_near_mid_point, quant_lens
+
+
+def check_outputs(arr_a: np.array, arr_b: np.array, is_near_mid_point: np.array, quant_lens: np.array, atol=0.000001):
```

Comment:
This file looks like a mixture of various helper functions. Some of them is used in tests but some in core functionality. It should be split and called appropriately.

Reply:
There is all functions used in tests. `check_outputs` to compare output, generators to generate test input. 
And `check_outputs` require outputs of generators from this file, because required additional arguments (is_near_mid_point. quant_lens). In my mind, it's looks like parts of one logic and should be in one file.